CRM/Contribute/Page/Tab.php

```diff
@@ -153,10 +153,11 @@ public function browse() {
     if (!empty($softCreditList)) {
       $softCreditTotals = array();
 
-      list($softCreditTotals['amount'],
+      list($softCreditTotals['count'],
+        $softCreditTotals['cancel']['count'],
+        $softCreditTotals['amount'],
         $softCreditTotals['avg'],
-        $softCreditTotals['currency'],
-        $softCreditTotals['cancelAmount'] // to get cancel amount
+        $softCreditTotals['cancel']['amount'] // to get cancel amount
```

Comment:
@pfigel when I looked at the other parts of this summary I concluded that flattening out the assigned array would be better - ie. we should flatten out the other arrays to be more like this one & then flatten again so we just iterate through one array - ie. the tpl would look more like

- in this context I think extracting the tpl change below makes is more separate from the main array rather than closer to being integrated

Reply:
although I can see your 2 template point!

Reply:
@eileenmcnaughton I'm thinking the way I'd go about refactoring the formatting stuff out of BAO (for soft credits and contribution totals/annuals) would be to return an array for `amount`, `cancelAmount` etc., i.e. something like
```php
$softCreditTotals = [
  'amount' => [
    'USD' => 10.00,
    'EUR' => 30.00,
  ]
]
```

Then we can just iterate over that and format it with `crmMoney`, plus we'd be consistent with array nesting levels.

Is this something we should do as part of this fix or separately? I'm happy to do it, just a bit uncertain on whether we should do a minimum viable fix if this goes a 5.10.x release, or if we should just go with the cleaner approach.

Reply:
@pfigel so at this stage we should target the 5.11 rc. We should fix all known regressions in 5.11 which is due out next Wed. (we may also do another 5.10 point drop - we generally don't this close to the next release but that q is open)

regarding the changes - I wasn't really proposing a larger refactoring - just thinking the cleanup you are doing is going in a different direction to how I was thinking we should clean it up in that you are separating Soft Credits out more & I was thinking about making it more so that the array assigned to the template provided the label & values & the template just rendered them. 

However, I agree - we should just fix it in a way that works at the moment - if you put this against the 5.11 rc & it works then I think we can merge

Reply:
Ah, I see. Yeah, I was going back and forth a bit myself while implementing this. Will leave it like this for now and keep the refactoring for another day.

Reply:
Yeah I agree this might not be the highest priority area for refactoring. I delved into it for performance reasons & now am trying to deal with extraneous search joins for the same reason